src/text/parsers/containers.rs

```diff
@@ -111,23 +110,28 @@ pub(crate) fn s_expression_value(input: &str) -> IResult<&str, AnnotatedTextValu
 /// Matches a (possibly annotated) non-container value in an s-expression followed by a delimiter
 /// or end-of-container.
 pub(crate) fn s_expression_scalar(input: &str) -> IResult<&str, AnnotatedTextValue> {
-    delimited(
+    preceded(
         whitespace_or_comments,
         // An s-expression value can be either...
         alt((
             // ...an annotated operator (`foo::++`)...
             pair(parse_annotations, parse_operator)
                 .map(|(annotations, value)| AnnotatedTextValue::new(annotations, value)),
-            // ...an un-annotated operator (`++`) paired with an empty annotations Vec...
-            parse_operator.map(|op| op.without_annotations()),
-            // ...or some other kind of value (`5`, `"hello"`, etc).
+            // ...a non-operator value, with or without annotations (`5`, `foo::5`, `"hello"`, etc)...
             top_level_value,
+            // ...or an un-annotated operator (`++`).
+            parse_operator.map(|op| op.without_annotations()),
```

Comment:
🗺️ See the comment starting on line 125 for an explanation of why this was shuffled.